velox/functions/lib/aggregates/AverageAggregateBase.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/lib/aggregates/AverageAggregateBase.h"
+
+namespace facebook::velox::functions::aggregate {
+
+void checkSumCountRowTypeForAvg(const TypePtr& type) {
+  const std::string errorMessage =
+      "Input type for final aggregation must be (sum:double/long decimal, count:bigint) struct";
```

Comment:
This message seems incorrect as input type for final agg can also be a varbinary.

Reply:
Improved the error messages.